File: windows/db_explorer/modules/drivers/INativeDatabaseDriver.cs
using db_explorer.modules.drivers.models;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace db_explorer.modules.drivers
{
    public interface INativeDatabaseDriver : IDisposable
    {
        Task<int> Create(Dictionary<string, string> connectionInfo);
        Task Connect(int id);
        Task Close(int id);
        Task<DatabaseQueryResult> Execute(int id, string sql, Dictionary<string, object> variables = null);
        Task Flush();
    }
}

File: src/database/native/INativeDatabaseDriver.ts
import DatabaseConnectionInfo from '../models/DatabaseConnectionInfo';
import DatabaseQueryResult from '../models/DatabaseQueryResult';

export default interface INativeDatabaseDriver {
  create(connectionInfo: DatabaseConnectionInfo): Promise<number>;
  connect(id: number): Promise<void>;
  close(id: number): Promise<void>;
  execute(
    id: number,
    sql: string,
    variables?: Record<string, any>,
  ): Promise<number>;
  query(
    id: number,
    sql: string,
    variables?: Record<string, any>,
  ): Promise<DatabaseQueryResult>;
  flush(): Promise<void>;
}
